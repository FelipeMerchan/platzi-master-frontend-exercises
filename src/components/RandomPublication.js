import React from 'react'

import './styles/RandomPublication.css'
import randomImage from '../images/cat.jpg'

function RandomPublication() {
  return (
    <section>
      <article className="RandomPublication skin">
        <figure className="RandomPublication__imageContainer">
          <img className="RandomPublication__image" src={randomImage} alt="Gato" />
        </figure>
        <div className="RandomPublication__details">
          <p className="RandomPublication__text">
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, 
            sed diam nonummy nibh euismod tincidunt ut laoreet.
          </p>
        </div>
      </article>
    </section>
  )
}

export default RandomPublication