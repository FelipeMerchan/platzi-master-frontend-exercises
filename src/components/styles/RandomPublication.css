.RandomPublication {
  display: flex;
  padding: var(--space-2);
  border-radius: var(--border-radius);
  font-family: var(--font-2);
  color: var(--typographyColor);
  background-color: var(--surfaceColor);
}

.RandomPublication__imageContainer {
  width: 30%;
  margin-right: 2%;
}

.RandomPublication__image {
  width: 100%;
}

.RandomPublication__details {
  width: 68%;
}

.RandomPublication__text {
  line-height: 1.2;
}

@media screen and (max-width: 768px) {
  .RandomPublication {
    display: block;
  }

  .RandomPublication__details,
  .RandomPublication__imageContainer {
    width: auto;
  }

  .RandomPublication__imageContainer {
    margin-bottom: var(--space);
    margin-right: 0;
  }
}