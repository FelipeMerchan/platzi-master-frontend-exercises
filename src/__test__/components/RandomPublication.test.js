import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import RandomPublication from '../../components/RandomPublication'

describe('Probar el componente <RandomPublication />', () => {
  const randomPublication = shallow(<RandomPublication />)

  test('Render del componente RandomPublication', () => {
    expect(randomPublication.length).toEqual(1)
  })
  test('Render de la imagen y los detalles', () => {
    expect(randomPublication.find('article').children()).toHaveLength(2);
  });
  test('Render del pÃ¡rrafo', () => {
    expect(randomPublication.find('p').text()).toEqual('Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet.')
  })
  test('Render con el className RandomPublication', () => {
    expect(randomPublication.exists('.RandomPublication')).toBeTruthy()
  })
})

describe('RandomPublication Snapshot', () => {
  test('Comprobar la UI del componente RandomPublication', () => {
    const randomPublication = create(<RandomPublication />)
    expect(randomPublication.toJSON()).toMatchSnapshot()
  })
})