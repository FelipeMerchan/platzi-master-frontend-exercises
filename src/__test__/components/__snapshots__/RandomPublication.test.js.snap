// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomPublication Snapshot Comprobar la UI del componente RandomPublication 1`] = `
<section>
  <article
    className="RandomPublication skin"
  >
    <figure
      className="RandomPublication__imageContainer"
    >
      <img
        alt="Gato"
        className="RandomPublication__image"
        src="cat.jpg"
      />
    </figure>
    <div
      className="RandomPublication__details"
    >
      <p
        className="RandomPublication__text"
      >
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet.
      </p>
    </div>
  </article>
</section>
`;
